File: vademecum/code-src/podstawy/Podstawowe_operacje_arytmetyczne_bitowe_i_logiczne.js

var a = 12.7, b = 3, c = 13, d, e;

// dodawanie, mnożenie, odejmowanie zapisuje się
// i działają one tak jak w normalnej matematyce:
e = (a + b) * 4 - c;

// dzielenie zawsze jest zmiennoprzecinkowe
d = a / b;
c = b / c;

// wypisanie wyników
console.log(
	"e=" + e + "\n" +
	"d=" + d + "\n" +
	"c=" + c + "\n"
);

// operacje logiczne:
// ((a większe równe od 0) AND (b mniejsze od 2)) OR (c równe 5)
var z = (a>=0 && b<2) || c == 5;
// negacja logiczna z
a = !z;

console.log(
	"z=" + z + "\n" +
	"a=" + a + "\n"
);

	// operacje binarne:
	// bitowy OR 0x0f z 0x11 i przesunięcie wyniku o 1 w lewo
	a = (0x0f | 0x11) << 1;
	// bitowy XOR 0x0f z 0x11
	b = (0x0f ^ 0x11);
	// negacja bitowa wyniku bitowego AND 0xfff i 0x0f0
	c = ~(0xfff & 0x0f0);
	
console.log(
	"a=0x" + a.toString(16) + "\n" +
	"b=0x" + b.toString(16) + "\n" +
	"c=0x" + (c & 0xffff).toString(16) + "\n"
);

File: vademecum/code-src/podstawy/Podstawowe_operacje_arytmetyczne_bitowe_i_logiczne.py

a = 12.7
b = 3
x = 5
y = 6

# dodawanie, mnożenie, odejmowanie zapisuje się i działają
# tak jak w normalnej matematyce:
e = (a + b) * 4 - y

# dzielenie zmiennoprzecinkowe
c = x / y

# dzielenie całkowite
b = a // b

# reszta z dzielenia
z = x % y;

# wypisanie wyników
print(e, c, b, z)

# operacje logiczne:
# ((a większe równe od 0) AND (b mniejsze od 2)) OR (z równe 5)
z = (a>=0 and b<2) or z == 5;
# negacja logiczna z
x = not z;

print(z, x);

# operacje binarne:
# bitowy OR 0x0f z 0x11 i przesunięcie wyniku o 1 w lewo
x = (0x0f | 0x11) << 1;
# bitowy XOR 0x0f z 0x11
y = (0x0f ^ 0x11);
# negacja bitowa wyniku bitowego AND 0xfff i 0x0f0
z = ~(0xfff & 0x0f0);

print(hex(x), hex(y), hex(z & 0xffff));
# wypisując z musimy określić jego bitowość

# wieloargumentowa operacja przypisania
a, b, c = 1, 2, 3
# Najpierw oblicza wartości wyrażeń po prawej, a dopiero potem przypisuje.
# Pozwala to np. na:
a, b = b, a
# celem zamiany wartości zmiennych bez jawnego używania zmiennej tymczasowej
print(a, b)
